src/runtime/pythonengine.cs

```diff
@@ -567,6 +576,54 @@ public static void Exec(string code, IntPtr? globals = null, IntPtr? locals = nu
             }
         }
 
+        /// <summary>
+        /// Gets the native thread ID.
+        /// </summary>
+        /// <returns>The native thread ID.</returns>
+        public static ulong GetNativeThreadID()
+        {
+            if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+            {
+                return GetCurrentThreadId();
+            }
+
+            if (RuntimeInformation.IsOSPlatform(OSPlatform.Linux))
+            {
+                return pthread_selfLinux();
+            }
+
+            if (RuntimeInformation.IsOSPlatform(OSPlatform.OSX))
+            {
+                return pthread_selfOSX();
+            }
+
+            throw new InvalidOperationException("Could not retrieve native thread ID.");
+        }
```

Comment:
I don't think this belongs to `PythonEngine`. From what I read, for the test you might be able to use `threading.get_ident()` from Python `thread` module.

Reply:
`get_ident` is a separate thing, the ODs are not compatible with the "remote raise".

Reply:
@filmor are you sure? Documentation is very bad on this, but here's what I found:

`get_ident` [appears to be](https://github.com/python/cpython/blob/6104013838e181e3c698cb07316f449a0c31ea96/Modules/_threadmodule.c#L1208) returning `PyThread_get_thread_ident`
`tstate->thread_id` [is also](https://github.com/python/cpython/blob/41010184880151d6ae02a226dbacc796e5c90d11/Python/pystate.c#L625) `PyThread_get_thread_ident()`

`PyThreadState_SetAsyncExc` [finds thread by its `tstate->thread_id`](https://github.com/python/cpython/blob/41010184880151d6ae02a226dbacc796e5c90d11/Python/pystate.c#L1129)

Reply:
Using get_ident seems to work. What would you like me to do? Modify GetNativeThreadID method or remove it completely?

Reply:
Remove it would be better.

Reply:
@lostmsu You are right. But then we should still expose `threading.get_ident` or `PyThread_get_thread_ident`, no?

Reply:
Why? The user can do what tests do now - e.g. call `Py.Import("threading").Invoke("get_ident")`.

@gpetrou It might make sense to mention `threading.get_ident()` in the summary of XML doc for the `Interrupt` method.

Reply:
I am getting mixed signals between the two of you :) Who is the BDFL for Python.NET? I added the get_ident usage in GetNativeThreadID for now.

Reply:
@gpetrou @filmor  well, I am voting for removal, as it is one less thing to support with potential multiple meanings, and having nothing to do with Python embedding on itself.

Reply:
@filmor ?